src/Parser/Mage/Frost/Modules/Features/BrainFreeze.js

```diff
@@ -11,81 +11,123 @@ const debug = false;
 // Brain Freeze appears to always fall after Flurry cast, but not always on same timestamp. Giving a margin here.
 const PROC_WINDOW_MS = 100;
 
+// When Glacial Spike is talented, it's OK to overwrite Brain Freeze procs if you have at least this many icicles.
+const ICICLE_BREAKPOINT = 3;
```

Comment:
I would have to look at events to figure out the order of things, but I would think it would only be acceptable for Brain Freeze to be refreshed if they have 3 or 4 icicles, not 1,2, or 5. Because at 5 they should be casting Glacial Spike if they have Brain Freeze and not Frostbolt (which can give more icicles).

Reply:
Very true! But wouldn't that be more appropriately handled in the Glacial Spike module? Since it's a mistake to not be casting GS at that point, the BF proc munching is more of a side-effect?

Reply:
it would technically be both. 

It is a overwrittenproc and therefore was a waste, and its a bad usage of Glacial Spike. Just because its a bad usage of Glacial Spike doesnt mean they didnt waste a Brain Freeze proc.

Reply:
Actually now that I am thinking about it, it's slightly more complicated. Because I'm pretty sure you are not supposed to cancel your FB cast even if you get your BF proc while you have 5 icicles already (sub 100ms or so). So maybe for now it's simpler to just keep the 3 cutoff and figure out the best way to handle the event ordering with the procs w/ five icicles and getting the proc at the same time as your fifth icicle?

Reply:
Fair. I think its technically better to cancel the Frostbolt to cast GlacialSpike but I would have to talk to the Theorycrafters in the Mage Discord about it, but for now we can probably count a refresh at 5 stacks as ok

Reply:
Later on I will work on adding the specific logic about checking to see when you got the proc. For more immediate concerns, I just realized that maybe 3 isn't the right number to use for the icicle cutoff. Because you get the icicle before the onRefresh I believe, (if BF procs) technically that means if you have 3 icicles when BF onRefreshes, you had 2 icicles when casting the FB. So that means the cutoff should be 4 because that would mean you had 3 when starting casting the FB. Correct me if I'm wrong there, but I think that's how the events go

Reply:
You should be able to look at the events tab to see the order that it happens.

Though that might not be the worst case, cause i think the guides say 3 because if you have 2 icicles and are in the middle of casting Frostbolt (i.e. you will have 3 at the end of the cast), then you hold Brain Freeze.

Reply:
Ok I checked, it looks like when applying a fresh buff it is on the same timestamp but before the icicle. When refreshing, brain freeze applies slightly after icicles (~150ms), probably due to the forgiveness window that allows you to flurry after frost bolt without munching. So I think I'll use the buffer option you mentioned to check and see how many icicles there were during the FB cast instead. I also double checked the mage discord and the consensus is you cast flurry->IL if you have 0, 1, or 2 icicles DURING the FB cast. The altered-time guide is worded incorrectly according to them. Luckily if they make up their minds and its different it should be pretty easy to change that const value